managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -285,6 +285,37 @@ public MarkDeleteEntry(PositionImpl newPosition, Map<String, Long> properties,
         return lastMarkDeleteEntry != null ? lastMarkDeleteEntry.properties : Collections.emptyMap();
     }
 
+    @Override
+    public boolean putProperty(String key, Long value) {
+        if (lastMarkDeleteEntry != null) {
+            MarkDeleteEntry currentLastMarkDeleteEntry = lastMarkDeleteEntry;
+            Map<String, Long> properties = currentLastMarkDeleteEntry.properties;
+            if (properties == null || properties.isEmpty()) {
+                Map<String, Long> newProperties = Maps.newHashMap();
+                newProperties.put(key, value);
+                lastMarkDeleteEntry = new MarkDeleteEntry(currentLastMarkDeleteEntry.newPosition, newProperties,
+                        currentLastMarkDeleteEntry.callback, currentLastMarkDeleteEntry.ctx);
+                lastMarkDeleteEntry.callbackGroup = currentLastMarkDeleteEntry.callbackGroup;
+            } else {
+                properties.put(key, value);
+            }
+            return true;
+        }
+        return false;
```

Comment:
I do not understand this part of the story.
Probably I miss some context.

if `lastMarkDeleteEntry` is null, then the command in finally ignored, we are not storing anyway the information requested by the user.

if this is correct, then I believe that we must fail the API call, because otherwise the user will think that the change is applied but this is not true.

Can you please clarify ?

Reply:
Fixed to return an error response when `lastMarkDeleteEntry` is null. However, if an instance of `PersistentSubscription` exists, the cursor should have already been initialized and `lastMarkDeleteEntry` should be set to a non-null value. So the REST API won't actually return that error.